src/main/java/storage/Storage.java

```diff
@@ -0,0 +1,131 @@
+     * @param ui     The Ui instance to handle user interface display.
+     * @param parser The Parser instance to interpret the commands.
+     * @throws LoadFileErrorException if the file cannot be found or an error occurs during loading.
+     */
+    public void loadTaskList(TaskList tasks, Ui ui, Parser parser) throws LoadFileErrorException {
+        try {
+            File saveFile = new File(FILE_PATH);
+            Scanner fileScanner = new Scanner(saveFile);
+            int currentTask = 0;
+            while (fileScanner.hasNextLine()) {
+                String line = fileScanner.nextLine().trim();
+                boolean isTaskDone = parser.getTaskDone(line);
+                String userCommandString = parser.getCommandString(line);
+                Command userCommand = parser.getCommand(userCommandString, false);
+                userCommand.execute(tasks, ui, this);
+                tasks.updateTask(isTaskDone, currentTask);
+                currentTask++;
+            }
+            ui.listTasks(tasks);
+        } catch (IndexOutOfBoundsException |
+                 FileNotFoundException | BentoException e) {
+            throw new LoadFileErrorException();
+        }
+    }
+
+    /**
+     * Saves tasks from the provided TaskList into the save file, without printing success message.
+     *
+     * @param tasks  The TaskList containing tasks to be saved.
+     * @param ui     The Ui instance to handle user interface display.
+     * @param parser The Parser instance to interpret the commands.
+     * @throws SaveFileErrorException if the file cannot be written or an error occurs during saving.
+     */
+    public void saveTaskListQuiet(TaskList tasks, Ui ui, Parser parser) throws SaveFileErrorException {
+        try {
+            // Create data directory if it does not exist
+            Files.createDirectories(DATA_DIRECTORY);
+            FileWriter saveWriter = new FileWriter(FILE_PATH);
+            for (Task task : tasks.getTasks()) {
+                saveWriter.write(task.getTaskAsCommand());
+                saveWriter.write(TASK_STATUS_DELIMITER);
+                writeTaskStatus(task, saveWriter);
+                saveWriter.write(System.lineSeparator());
+            }
+            saveWriter.close();
+            // ui.printSaveTaskListSuccessMessage();
+        } catch (IOException e) {
+            throw new SaveFileErrorException();
+        }
+    }
+
+    /**
+     * Saves tasks from the provided TaskList into the save file and prints a success message.
+     *
+     * @param tasks  The TaskList containing tasks to be saved.
+     * @param ui     The Ui instance to handle user interface display.
+     * @param parser The Parser instance to interpret the commands.
+     * @throws SaveFileErrorException if the file cannot be written or an error occurs during saving.
+     */
+    public void saveTaskList(TaskList tasks, Ui ui, Parser parser) throws SaveFileErrorException {
```

Comment:
The saveTaskList and saveTaskListQuiet methods contain almost identical logic. You can try to extract the common logic into a private method and pass a boolean to indicate whether to print the success message.